assets/chat/js/formatters/UrlFormatter.js

```diff
@@ -72,14 +73,23 @@ export default class UrlFormatter {
       if (m) {
         const encodedUrl = self.encodeUrl(m[0]);
         const normalizedUrl = this.normalizeUrl(encodedUrl);
+
+        let embedHashLink = '';
+        try {
+          embedHashLink = this.hashLinkConverter.convert(normalizedUrl);
+        } catch (err) {
+          // ignore
+        }
         const maxUrlLength = 90;
         let urlText = normalizedUrl;
         if (urlText.length > maxUrlLength) {
           urlText = `${urlText.slice(0, 40)}...${urlText.slice(-40)}`;
         }
         const extra = self.encodeUrl(decodedUrl.substring(m[0].length));
         const href = `${scheme ? '' : 'http://'}${normalizedUrl}`;
-        return `<a target="_blank" class="externallink ${extraclass}" href="${href}" rel="nofollow">${urlText}</a>${extra}`;
+        return embedHashLink
+          ? `<a target="_blank" class="externallink ${extraclass}" href="${href}" rel="nofollow">${urlText}</a><a class="embed-button" href="/bigscreen${embedHashLink}"><svg xmlns="http://www.w3.org/2000/svg" width="18" height="14.4" viewBox="0 0 640 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc. --><path d="M64 64V352H576V64H64zM0 64C0 28.7 28.7 0 64 0H576c35.3 0 64 28.7 64 64V352c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V64zM128 448H512c17.7 0 32 14.3 32 32s-14.3 32-32 32H128c-17.7 0-32-14.3-32-32s14.3-32 32-32z"  fill="#fff"/></svg></a>`
```

Comment:
No `target` on this anchor means that, when the chat is embedded in an iframe, the stream content is added to that iframe. Merge in master and check out `EmbedUrlFormatter` for how to set this up properly.

Reply:
I've added in the `const target = chat.isBigscreenEmbed() ? '_top' : '_blank';` line and used that in the anchor link for the embed is there anything additional needed?